### PR TITLE
PYTHON-3311 Module "pymongo" does not explicitly export attribute "MongoClient"; implicit reexport disabled

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Review by ShaneHarvey — COMMENTED** [review]
Nice test! One concern, is adding `__all__` a breaking change for any users?

File: pymongo/database.py [hunk 1]
```diff
@@ -699,7 +700,7 @@ def command(
         check: bool = True,
         allowable_errors: Optional[Sequence[Union[str, int]]] = None,
         read_preference: Optional[_ServerMode] = None,
-        codec_options: "Optional[CodecOptions[_CodecDocumentType]]" = None,
+        codec_options: "Optional[bson.codec_options.CodecOptions[_CodecDocumentType]]" = None,
```
[the remark this answers is not in the document]
Comment (blink1073) [reply pymongo/database.py:703]: Sphinx saw an ambiguity between `bson.CodecOptions` and `bson.codec_options.CodecOptions`.  ¯\_(ツ)_/¯

File: test/test_default_exports.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+GLOBAL_INGORE = ["TYPE_CHECKING"]
+
+
+class TestDefaultExports(unittest.TestCase):
+    def check_module(self, mod, ignores):
+        names = dir(mod)
+        names.remove("__all__")
+        for name in mod.__all__:
+            if name not in names and name not in ignores:
+                raise ValueError(name)
```
[the remark this answers is not in the document]
Comment (blink1073) [reply test/test_default_exports.py:45]: Done

**Comment by ShaneHarvey** [comment]
Still have this question:
> is adding __all__ a breaking change for any users?

**Comment by blink1073** [comment]
> is adding __all__ a breaking change for any users?

If they were relying on any of the items in `GRIDFS_IGNORE` it would be a breaking change.  We could choose to leave it as an empty list to be extra conservative.  I chose only ones that weren't from `GridFS` itself.

**Comment by blink1073** [comment]
Also, if the user was relying on any of the `typing` imports on `import *` that would be a breaking change.

**Review by ShaneHarvey — APPROVED** [review]
Cool, LGTM!

**Pull request merged**